projects/com.oracle.truffle.llvm.frontend/src/com/oracle/truffle/llvm/frontend/bc/LLVMBitcodeHelper.java

```diff
@@ -0,0 +1,973 @@
+                case I8:
+                    return LLVMRuntimeType.I8_VECTOR;
+                case I16:
+                    return LLVMRuntimeType.I16_VECTOR;
+                case I32:
+                    return LLVMRuntimeType.I32_VECTOR;
+                case I64:
+                    return LLVMRuntimeType.I64_VECTOR;
+                case FLOAT:
+                    return LLVMRuntimeType.FLOAT_VECTOR;
+                case DOUBLE:
+                    return LLVMRuntimeType.DOUBLE_VECTOR;
+                default:
+                    throw new RuntimeException("Unsupported type " + type);
+            }
+        }
+        throw new RuntimeException("Unsupported type " + type);
+    }
+
+    public static LLVMRuntimeType[] toRuntimeTypes(Type[] types) {
+        LLVMRuntimeType[] llvmtypes = new LLVMRuntimeType[types.length];
+
+        for (int i = 0; i < types.length; i++) {
+            llvmtypes[i] = toRuntimeType(types[i].getType());
+        }
+
+        return llvmtypes;
+    }
+
+    public static LLVMRuntimeType[] toRuntimeTypes(List<? extends Type> types) {
+        LLVMRuntimeType[] llvmtypes = new LLVMRuntimeType[types.size()];
+
+        for (int i = 0; i < types.size(); i++) {
+            llvmtypes[i] = toRuntimeType(types.get(i).getType());
+        }
+
+        return llvmtypes;
+    }
+
+    private static LLVMAddressNode toArray(Type type, int alignment, List<LLVMExpressionNode> values, LLVMContext context, FrameSlot stack) {
+        LLVMBaseType llvmElementType = toBaseType(type);
+        int baseTypeSize = type.sizeof();
+        int nrElements = values.size();
+        int size = nrElements * baseTypeSize;
+        if (size == 0) {
+            throw new AssertionError(llvmElementType + " has size of 0!");
+        }
+        LLVMAllocaInstruction target = LLVMAllocaInstructionNodeGen.create(size, alignment, context, stack);
+        switch (llvmElementType) {
+            case I8:
+                return LLVMStoreNodeFactory.LLVMI8ArrayLiteralNodeGen.create(values.toArray(new LLVMI8Node[nrElements]), baseTypeSize, target);
+            case I16:
+                return LLVMStoreNodeFactory.LLVMI16ArrayLiteralNodeGen.create(values.toArray(new LLVMI16Node[nrElements]), baseTypeSize, target);
+            case I32:
+                return LLVMStoreNodeFactory.LLVMI32ArrayLiteralNodeGen.create(values.toArray(new LLVMI32Node[nrElements]), baseTypeSize, target);
+            case I64:
+                return LLVMStoreNodeFactory.LLVMI64ArrayLiteralNodeGen.create(values.toArray(new LLVMI64Node[nrElements]), baseTypeSize, target);
+            case FLOAT:
+                return LLVMStoreNodeFactory.LLVMFloatArrayLiteralNodeGen.create(values.toArray(new LLVMFloatNode[nrElements]), baseTypeSize, target);
+            case DOUBLE:
+                return LLVMStoreNodeFactory.LLVMDoubleArrayLiteralNodeGen.create(values.toArray(new LLVMDoubleNode[nrElements]), baseTypeSize, target);
+//            case X86_FP80:
```

Comment:
I have only looked at the su-tests-sulong so far so for integration. The half, x86_fp80, fp128 and ppc_fp128 data types are supported in the decoder but as I have had no need to integrate it or test them, the fp80 remains commented out.